ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -338,58 +362,96 @@ runWith RunNodeArgs{..} LowLevelRunNodeArgs{..} =
-          rnTraceNTC
-          (NTC.defaultCodecs codecConfig blockVersion networkVersion)
-          (NTC.mkHandlers nodeKernelArgs nodeKernel)
+      NTC.mkApps
+        nodeKernel
+        rnTraceNTC
+        (NTC.defaultCodecs codecConfig blockVersion networkVersion)
+        (NTC.mkHandlers nodeKernelArgs nodeKernel)
 
     mkDiffusionApplications
-      :: MiniProtocolParameters
-      -> (   BlockNodeToNodeVersion blk
-          -> NTN.Apps m (ConnectionId addrNTN) ByteString ByteString ByteString ByteString ByteString ()
-         )
-      -> (   BlockNodeToClientVersion blk
+      :: NetworkP2PMode
+      -> MiniProtocolParameters
+      -> (BlockNodeToNodeVersion blk
+          -> NTN.Apps
+               m
+               (ConnectionId ntnAddr)
+               ByteString
+               ByteString
+               ByteString
+               ByteString
+               ByteString
+               ())
+      -> (BlockNodeToClientVersion blk
           -> NodeToClientVersion
-          -> NTC.Apps m (ConnectionId addrNTC) ByteString ByteString ByteString ()
-         )
-      -> NodeKernel m (ConnectionId addrNTN) (ConnectionId addrNTC) blk
-      -> PeerMetrics m addrNTN
+          -> NTC.Apps
+               m (ConnectionId ntcAddr) ByteString ByteString ByteString ())
+      -> NodeKernel m remotePeer localPeer blk
+      -> PeerMetrics m ntnAddr
       -> BlockchainTime m
       -> DiffusionApplications
-           addrNTN addrNTC
-           versionDataNTN versionDataNTC
+           ntnAddr
+           ntcAddr
+           versionDataNTN
+           versionDataNTC
            m
-    mkDiffusionApplications miniProtocolParams ntnApps ntcApps kernel peerMetrics btime =
-      DiffusionApplications {
-          daApplicationInitiatorMode = combineVersions
-             [ simpleSingletonVersions
-                 version
-                 llrnVersionDataNTN
-                 (NTN.initiator miniProtocolParams version $ ntnApps blockVersion)
-             | (version, blockVersion) <- Map.toList llrnNodeToNodeVersions
-             ]
-        , daApplicationInitiatorResponderMode = combineVersions
-             [ simpleSingletonVersions
-                 version
-                 llrnVersionDataNTN
-                 (NTN.initiatorAndResponder miniProtocolParams version $ ntnApps blockVersion)
-             | (version, blockVersion) <- Map.toList llrnNodeToNodeVersions
-             ]
-        , daLocalResponderApplication = combineVersions [
-              simpleSingletonVersions
+    mkDiffusionApplications
+      enP2P
+      miniProtocolParams
+      ntnApps
+      ntcApps
+      kernel
+      peerMetrics
+      btime =
+      case enP2P of
+        EnabledP2PMode ->
+          mkDiffusionApplicationsP2P
+            initiator
+            initiatorAndResponder
+            responder
+            ledgerPeersConsensusInterface
+            miniProtocolParams
+            (consensusRethrowPolicy (Proxy @blk))
+            mempty
```

Comment:
Would you add a comment here, explain what `mempty` is?

Or better yet, create a `where` binding

```
localRethrowPolicy :: RethrowPolicy
localRethrowPolicy = mempty
```